roles/approve_add_host/tasks/main.yml

```diff
@@ -0,0 +1,16 @@
+- name: Get Agent name
+  shell: "oc get agent -n {{ namespace }} --kubeconfig {{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}} -o json | jq -r .items[].metadata.name"
+  register: oc_get_agent_name
+
+- name: Approve the host
+  shell: "oc patch agent {{ oc_get_agent_name.stdout }} -n {{ namespace }} --kubeconfig {{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}} --type merge --patch '{\"spec\":{\"approved\":true }}'"
+
+- name: Add host to cluster
+  shell: "oc patch agent {{ oc_get_agent_name.stdout }} -n {{ namespace }} --kubeconfig {{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}} --type merge --patch '{\"spec\":{\"clusterDeploymentName\":{\"name\": \"{{ namespace }}\", \"namespace\": \"{{ namespace }}\"}}}'"
+
+- name: Wait until Host is installed
+  command: "oc get agent {{ oc_get_agent_name.stdout }} -n {{ namespace }} --kubeconfig {{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}} -o json"
+  register: oc_get_managedcluster
+  until: oc_get_managedcluster.stdout|from_json|json_query('status.conditions[*].status')|unique == ["True"]

```

Comment:
Lets work together on this after the new year I think we can refactor this to use the kubenetes module, less shell and command calls is a much better way of doing ansible. 

Previously we used oc as it was only a few commands and we didn't see the need to add a new dependency. I think we have far gone passed the threshold where a new dependency is worth it :)